test/dlc_tests/conftest.py

```diff
@@ -276,6 +277,37 @@ def example_only():
     pass
 
 
+@pytest.fixture(scope="session")
+def tf2_only():
+    pass
+
+
+@pytest.fixture(scope="session")
+def tf23_and_above_only():
+    pass
+
+
+@pytest.fixture(scope="session")
+def tf21_and_above_only():
+    pass
+
+
+def tf_version_within_limit(metafunc_obj, image):
+    """
+    Test all pytest fixtures for TensorFlow version limits, and return True if all requirements are satisfied
+
+    :param metafunc_obj: pytest metafunc object from which fixture names used by test function will be obtained
+    :param image: Image URI for which the validation must be performed
+    :return: True if all validation succeeds, else False
+    """
+    tf2_requirement_failed = "tf2_only" in metafunc_obj.fixturenames and is_tf1(image)
+    tf23_requirement_failed = "tf23_and_above_only" in metafunc_obj.fixturenames and below_tf23(image)
+    tf21_requirement_failed = "tf21_and_above_only" in metafunc_obj.fixturenames and (is_tf1(image) or is_tf20(image))
```

Comment:
tf23_requirement_failed and tf21_requirement_failed essentially are trying to accomplish the same thing with different tf versions, but are implemented differently. Let's unify this - can you create a general function for checking if tf version is below a certain threshold (i.e. `is_below_tf_version(image, tf_version)`) and call it in both cases?

Reply:
Yeah, I was thinking of doing this, but didn't want to go out of scope on this PR. I already have an implementation for this. Will update and submit another revision.